core/templates/dev/head/pages/exploration_editor/exploration_save_and_publish_buttons_directive.html

```diff
@@ -64,4 +64,18 @@
       </div>
     </li>
   </ul>
+  <script type="text/ng-template" id="modals/confirmDiscardChanges">    
+    <div class="modal-header">
+      <button type="button" class="close" ng-click="close(false)" data-dismiss="modal" aria-hidden="true">&times;</button>
+      <h4 class="modal-title">Confirmation Required</h4>
+    </div>
+    <div class="modal-body">
+      <p>Are you sure you want to discard the changes?</p>
+    </div>
+    <div class="modal-footer">
+      <button type="button" ng-click="close(false)" class="btn btn-default" data-dismiss="modal">Cancel</button>
```

Comment:
@1995YogeshSharma -- similar concerns here (with respect to using close() for both) as in #3055. Could you fix?

Also, what's data-dismiss for (here and below)? I don't see it in other modals.